retry/lib/retry.dart

```diff
@@ -120,8 +120,8 @@ class RetryOptions {
   /// as an [Exception].
   Future<T> retry<T>(
     FutureOr<T> Function() fn, {
-    FutureOr<bool> Function(Exception)? retryIf,
-    FutureOr<void> Function(Exception)? onRetry,
+    FutureOr<bool> Function(Exception, int)? retryIf,
```

Comment:
This is a breaking change for all the libraries that are using this method, and should be reflected in version and CHANGELOG as well. The method also needs more documentation on what the `int` means.

I'm wondering if there is a different signature that is not a breaking change, e.g. with a different callback parameter?

Reply:
Yeah, we could do: `FutureOr<boo> Function(Exception e, int attempt) retryAt`

But it doesn't super well, and IMO, I'm not sure there is a good way to limit retries based on the kind of exception thrown.
You won't remember if previous attempts failed due to some other `Exception`.